### DISCUSSION TITLE
How to obtain a pairingTopic?

### DISCUSSION THREAD

**Discussion by scryptoking** [WalletConnect/walletconnect-monorepo#2299] [Developer Support] [answered]
Hi All,

I can't get my head around how to obtain a `pairingTopic`. When I follow the guides on `ehtereum-provider` you can either use `provider.enable()` or `provider.connect({})`. Withing the object of `provider.connect` the docs state that you can use an already existing `pairingTopic`. Example from the [docs](https://docs.walletconnect.com/2.0/web/providers/ethereum#display-web3modal-with-qr-code--handle-connection-uri):

```
await provider.connect({
  chains, // OPTIONAL chain ids
  rpcMap, // OPTIONAL rpc urls
  pairingTopic // OPTIONAL pairing topic
})
```
When I first connect a wallet through the `provider.enable()` method, the `pairingTopic` key in the provider is `undefined` (`provider.signer.session.pairingTopic`).

My question thus is: how or where does one establish a `pairingTopic`?

Many thanks in advance!

**Comment by finessevanes** [comment, marked as answer]
You can get a list of pairings like this:
`const pairings = provider.signer.client.pairing.getAll({ active: true });`
https://github.com/WalletConnect/se-sdk/blob/1e8c766876fe5b5e31c3687d5465369e226f4866/examples/dapp/src/contexts/ClientContext.tsx#LL230C7-L231C1
